components/InfoBoxCluster.vue

```diff
@@ -0,0 +1,146 @@
+<script>
+import { isEmpty } from 'lodash';
+import InfoBox from '@/components/InfoBox';
+import PercentageBar from '@/components/PercentageBar';
+import Units from '@/utils/units';
+
+export default {
+  components: {
+    InfoBox,
+    PercentageBar,
+  },
+
+  props: {
+    cluster: {
+      type:     Object,
+      required: true,
+    },
+    metrics: {
+      type:     Array,
+      required: true,
+    }
+  },
+
+  computed: {
+    liveNodeUsage() {
+      const clusterCapacityCpu = this.cluster?.status?.capacity?.cpu;
+
+      return this.getUsage(clusterCapacityCpu, Units.parseSi(clusterCapacityCpu), this.metrics, 'cpu');
+    },
+
+    nodeUsageReserved() {
+      const allocatableCpu = this.cluster?.status?.allocatable?.cpu;
+
+      return this.getUsage(allocatableCpu, Units.parseSi(allocatableCpu), this.metrics, 'cpu');
+    },
+
+    liveNodeMemUsage() {
+      let clusterCapacityMemory = this.cluster?.status?.capacity?.memory;
+
+      clusterCapacityMemory = Units.parseSi(clusterCapacityMemory);
+
+      return this.getUsage(clusterCapacityMemory, clusterCapacityMemory, this.metrics, 'memory');
+    },
+
+    nodeUsageMemReserved() {
+      let allocatableMem = this.cluster?.status?.allocatable?.memory;
+
+      allocatableMem = Units.parseSi(allocatableMem);
+
+      return this.getUsage(allocatableMem, allocatableMem, this.metrics, 'memory');
+    },
+
+    nodeUsagePodReserved() {
+      const podCapacity = Units.parseSi(this.cluster?.status?.allocatable?.pods || 0);
+      const podUsage = Units.parseSi(this.cluster?.status?.requested?.pods || 0);
+
+      return {
+        clusterCapacity: Units.formatSi(podCapacity),
+        nodeUsage:       Units.formatSi(podUsage),
+        percentage:      ( podUsage * 100 ) / podCapacity,
+      };
+    },
+  },
+
+  methods: {
+    getUsage(clusterCapacityCpu = 0, normalizedCpu = 0, metrics = [], field = 'cpu') {
```

Comment:
Why are both `clusterCapacityCpu` and `normalizedCpu` passed in?  They're sometimes the same value, sometimes one is the raw string and the other parsed..  Just take in the raw value and parse it once inside getUsage().

Also this is used for more than just CPU...

Reply:
Yeah I'll fix that, the function evolved a bit and I hadn't gone back and optimized.